src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -26,25 +22,136 @@ authoritative if this occurs. Instead of pinging individual members, use
+
+This walkthrough will cover:
+
+1. Compiling a Rust binary targeting Fuchsia.
+1. Building a Fuchsia package.
+1. Publishing and running a Fuchsia package to a Fuchsia emulator.
+
+For the purposes of this walkthrough, we will only target `x86_64-fuchsia`.
+
+## Compiling a Rust binary targeting Fuchsia
+
+Today, there are two main ways to build a Rust binary targetting Fuchsia:
+
+1. Allowing [rustup] to handle the installation of Fuchsia targets for you,
+   and using the [cargo] tool to build the Rust binary.
+1. Using the Fuchsia SDK and related `sysroot` to build and install a `rustc`
+   that can target Fuchsia.
+
+### Compiling a Rust binary targeting Fuchsia (Rustup/Cargo)
+
+The easiest way to build a Rust binary targeting Fuchsia is by allowing [rustup]
+to handle the installation of Fuchsia targets for you. This can be done by issuing
+the following commands:
+
+```sh
+rustup target add x86_64-fuchsia
+rustup target add aarch_64-fuchsia
+```
+
+After installing our Fuchsia targets, we can now compile a Rust binary that targets
+Fuchsia.
+
+To create our Rust project, we can issue a standard `cargo` command as follows:
+
+**From base working directory**
+```sh
+cargo new \
+    --name hello_fuchsia \
+    --bin hello_fuchsia_src
+```
+
+We can edit our `hello_fuchsia_src/src/main.rs` to include a test as follows:
+
+**`hello_fuchsia_src/src/main.rs`**
+```rust
+fn main() {
+    println!("Hello Fuchsia!");
+}
+
+#[test]
+fn it_works() {
+    assert_eq!(2 + 2, 4);
+}
+```
+
+In addition to the standard workspace created, we will want to create a
+`hello_fuchsia_src/build.rs` file to link necessary libraries during compilation:
+
+**`hello_fuchsia_src/build.rs`**
+```rust
+fn main() {
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/sysroot/lib");
+  println!("cargo:rustc-link-search=native=<SDK_PATH>/arch/x64/lib");
+}
+```
+
+*Note: Make sure to fill out `<SDK_PATH>` with the path to the downloaded [Fuchsia SDK].*
+
+In total, our new project will look like:
+
+**Current directory structure**
+```txt
+hello_fuchsia_src/
+┣━ src/
+┃  ┗━ main.rs
+┣━ Cargo.toml
+┗━ build.rs
+```
+
+Finally, we can build our rust binary as:
+
+```sh
+cargo build \
+    --target x86_64-fuchsia \
+    --manifest-path hello_fuchsia_src/Cargo.toml
+```
+
+Now we have a Rust binary at `hello_fuchsia_src/target/x86_64-fuchsia/debug/hello_fuchsia`,
+targeting our desired Fuchsia target.
+
+Copy this binary into an intermediate directory, to be used in later steps:
+
+```sh
+mkdir hello_fuchsia_bin
+cp hello_fuchsia_src/target/x86_64-fuchsia/debug/hello_fuchsia \
+   hello_fuchsia_bin/hello_fuchsia
+```
+
+**Current directory structure**
+```txt
+hello_fuchsia_src/
+┣━ src/
+┃  ┗━ main.rs
+┣━ target/
+┃  ┣━ x86_64-fuchsia/
+┃  ┃  ┣━ debug/
+┗━━┻━━┻━━┻━ hello_fuchsia       <-
+hello_fuchsia_bin/               | Copied binary
+┗━ hello_fuchsia                <-
+```
+
+### Compiling a Rust binary targeting Fuchsia (from source)
```

Comment:
"from source" sounds to me like you're talking about the user code instead of compiler, maybe something like..

> Targeting Fuchsia with a compiler built from source